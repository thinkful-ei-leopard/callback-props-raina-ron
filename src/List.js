import React from 'react';

function List(prop) {
  return (
      <section className='List'>
          <header>
              {prop.header}
          </header>
          <div className='List-cards'>
            
          </div>
      </section>
  );
}


export default List;