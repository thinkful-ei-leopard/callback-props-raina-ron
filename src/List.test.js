import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('<List />', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<List cards={[{id: "a", title: "First card", content: "lorem ipsum"}]}  />, div);

        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected ', () => {
        const tree = renderer.create(<List cards={[{id: "a", title: "First card", content: "lorem ipsum"}]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});