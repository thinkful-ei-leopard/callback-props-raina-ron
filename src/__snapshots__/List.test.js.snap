// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected  1`] = `
<section
  className="List"
>
  <header />
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3 />
      <p />
    </div>
  </div>
</section>
`;
